src/function/scalar/list/list_aggregates.cpp

```diff
@@ -145,44 +210,136 @@ static void ListAggregateFunction(DataChunk &args, ExpressionState &state, Vecto
 		aggr.function.update(&slice, aggr.bind_info.get(), 1, state_vector_update, states_idx);
 	}
 
-	// finalize all the aggregate states
-	aggr.function.finalize(state_vector.state_vector, aggr.bind_info.get(), result, count, 0);
+	if (IS_AGGR) {
+		// finalize all the aggregate states
+		aggr.function.finalize(state_vector.state_vector, aggr.bind_info.get(), result, count, 0);
+
+	} else {
+		// finalize manually to use the map
+		D_ASSERT(aggr.function.arguments.size() == 1);
+		auto key_type = aggr.function.arguments[0];
+
+		switch (key_type.id()) {
```

Comment:
Why are we switching on `LogicalType` here (and in `histogram.cpp`) instead of `PhysicalType`? The behaviour should be the same regardless of `LogicalType`, and we are just using the `PhysicalType`, i.e., `int32_t`/`uint64_t`/etc. for the map.

Also I've just spotted that `UINT8` and`INT8` are missing (here and in `histogram.cpp`), could you add those as well?

Reply:
`histogram.cpp` did not have `UINT8` and `INT8` prior to my changes, but I can see if adding them adds any big/unexpected changes.

What would be the difference/benefit of switching between the `PhysicalType` instead? I see that it was done like that for some of the cases in the old histogram code. I was not sure which one to choose, also since they seem to have about the same amount of cases.

Reply:
We have less physical types than logical types, because some logical types use the same physical type, i.e., `DATE` uses `INT64` I believe. For these functions the logical type does not matter during execution, because we only care about the physical type anyway, we would not do anything differently if the logical type was different.

Reply:
Ok so I checked the physical types and made this list of which ones to include in the switch cases. For some of the types I am not completely sure and need a bit help.

**Not sure about these?**

- BOOL with `<uint8_t>`?
- HALF_FLOAT with `<float>`?
- STRING with `<string>`?
- LARGE_STRING with `<string>`?
- VARCHAR with `<string>`
- INT128 with ??

**These won't be included, right?**

- BINARY
- FIXED_SIZE_BINARY
- INTERVAL
- LIST
- STRUCT
- UNION
- DICTIONARY
- MAP
- EXTENSION
- FIXED_SIZE_LIST
- DURATION
- LARGE_BINARY
- LARGE_LIST
- UNKNOWN
- BIT
- INVALID

Reply:
> - BOOL with <uint8_t>?

Yes
> - HALF_FLOAT with <float>?

I don't think we support this yet, don't worry about this
> - STRING with <string>?
> - LARGE_STRING with <string>?
> - VARCHAR with <string>?

You should support `VARCHAR`/`BLOB` with `string_t`
> - INT128 with ??
With `hugeint_t`

Reply:
Ah sorry, `BOOL` is just `bool` actually, my bad!

Reply:
Ok and why `string_t` instead of `string`? The previous histogram implementation uses `string` for the map.